### 1. Use read_csv() to read the responses of a previous semesters entry poll from this source 
    a) Read the attached file “master_data_ready.csv” and assign it to the variable ‘ep’.
    b) How many variables and responses are there? What are these variables and responses in terms of the dataset?
    c) Display the datatypes of the variables.
    d) Change the variable names to something shorter.
    e) Describe what the following code does: ep[list(ep)[1]].value_counts().
    Suggestion: Break it into two or more statements and explain them one at a time.

In [2]:
# a) Read the attached file “master_data_ready.csv” and assign it to the variable ‘ep’
import pandas as pd
ep = pd.read_csv('entry_poll_fall16.csv')

In [3]:
# b) How many variables and responses are there? What are these variables and responses in terms of the dataset?
ep.shape

(85, 10)

ANS: 10 variables, with 85 responses. The variables seems like survey questions for this class (or similar class), and the 85 responses are the response of thoes survey questions.

In [4]:
# c) Display the datatypes of the variables.
ep.dtypes

TimeStamp                                                                                              object
Why have you enrolled in this course?                                                                  object
What program are you enrolled in?                                                                      object
How much programming experience have you had?                                                          object
How much experience with probability and statistics have you had?                                      object
What is your expected major or concentration?                                                          object
What career path most interests you?                                                                   object
Do you use social media for information purposes?  Check all that apply.                               object
What kinds of data most interest you?  List any that cross your mind.                                  object
If we have

All variables have data type "object".

In [5]:
# d) Change the variable names to something shorter.
ep = ep.rename(columns={"TimeStamp":"Time", "Why have you enrolled in this course?  ":"Reason for enrollment",
                  "What program are you enrolled in?":"Current Program", 
                   "How much programming experience have you had?":"Programming Experience",
                  "How much experience with probability and statistics have you had?":"Prob/Stat Experience",
                  "What is your expected major or concentration? ":"Major",
                  "What career path most interests you? ":"Interested Career Path",
                  "Do you use social media for information purposes?  Check all that apply.":"Social Media",
                  "What kinds of data most interest you?  List any that cross your mind.  ":"Interested Data",
                  "If we have time -- and we may not -- what special topics would interest you?  Check all that apply.":"Iterested Special Topic"})

In [6]:
ep.head(2)

Time   Reason for enrollment         Current Program  \
0  2016-08-16 12:55:02  To help with my career  Undergraduate business   
1  2016-08-31 05:54:00      I heard it was fun     Other undergraduate   

  Programming Experience                               Prob/Stat Experience  \
0                   None  I have taken one probability or statistics course   
1                   None                                               None   

                      Major            Interested Career Path Social Media  \
0  Analytics or other quant  Technology (Google, Amazon, etc)      Twitter   
1                       NaN                           Finance         None   

                                 Interested Data  \
0  Financial Data, Logistical Data, Housing Data   
1                                            NaN   

                      Iterested Special Topic  
0                Multivariate regression;Maps  
1  None, I'd prefer to focus on fundamentals.

In [7]:
# e) Describe what the following code does: ep[list(ep)[1]].value_counts(). 
# Suggestion: Break it into two or more statements and explain them one at a time.

ep[list(ep)[1]].value_counts()

To help with my career                               60
I heard it was fun                                   14
I lost my mind for a minute                           5
I code/build websites in my spare time as a hobby     1
To finish my major                                    1
i am living in Santiago de Chile                      1
Recommended by a professor.                           1
Wanted to try something completely different          1
I like a challenge                                    1
Name: Reason for enrollment, dtype: int64

In [8]:
list(ep)

['Time',
 'Reason for enrollment',
 'Current Program',
 'Programming Experience',
 'Prob/Stat Experience',
 'Major',
 'Interested Career Path',
 'Social Media',
 'Interested Data',
 'Iterested Special Topic']

First list(ep) turn columns name into a list.

In [9]:
list(ep)[1]

'Reason for enrollment'

Then it chooses the second index, as Python counts from 0, which is 'Reason for enrollment'.

In [10]:
ep[list(ep)[1]].head()

0    To help with my career
1        I heard it was fun
2        I like a challenge
3        I heard it was fun
4        I heard it was fun
Name: Reason for enrollment, dtype: object

Then the above code is same as ep['Reason for enrollement'], and it will show you all rows belong to this column.

In [11]:
ep[list(ep)[1]].value_counts()

To help with my career                               60
I heard it was fun                                   14
I lost my mind for a minute                           5
I code/build websites in my spare time as a hobby     1
To finish my major                                    1
i am living in Santiago de Chile                      1
Recommended by a professor.                           1
Wanted to try something completely different          1
I like a challenge                                    1
Name: Reason for enrollment, dtype: int64

Finally, value_counts() counts each unique values in that column.

### 2. Read the data set to your notebook. Modify the dataset to replace \N values everywhere in the dataset with NaN. Capitalize the first letter of each column name and assign the new names to the columns of the dataset.

In [12]:
df = pd.read_csv('master_data_ready.csv')

In [13]:
# Replace \N value with NaN everywhere.
import numpy as np
df = df.replace(r"\N", np.nan)

In [14]:
df.sample(1)

job_id                vertical      company  \
190975  3038107179395527583442890985562  Information Technology  Bosch Group   

         post_date fill_date time_to_fill  salary          location  \
190975  2018-02-23       NaN          NaN   62599  Walnut Ridge, AR   

                city state    zip    county             region_state  \
190975  Walnut Ridge    AR  72476  Lawrence  AR NONMETROPOLITAN AREA   

                 latitude           longitude  
190975  36.02439880371094  -90.92880249023438

In [15]:
# Caplitalize the first letter of each column name and assign new names to the columns of the dataset.
df.columns = [x.capitalize() for x in df.columns]

In [16]:
df.sample(2)

Job_id                Vertical Company  \
66088  42590097433571718901556615006321  Information Technology     NaN   
67645  44576019150408094147599462257544  Information Technology     NaN   

        Post_date   Fill_date Time_to_fill  Salary Location City State  Zip  \
66088  2016-09-10  2016-09-17            7   42849      NaN  NaN   NaN  NaN   
67645  2017-02-14         NaN          NaN   80047      NaN  NaN   NaN  NaN   

      County Region_state Latitude Longitude  
66088    NaN          NaN      NaN       NaN  
67645    NaN          NaN      NaN       NaN

### 3. Using an appropriate pandas method present a column wise summary of nulls (or NaN)

In [17]:
df.isna().sum()

Job_id               0
Vertical             0
Company         110408
Post_date            0
Fill_date       151118
Time_to_fill    151118
Salary               0
Location        110730
City            112673
State           110823
Zip             114932
County          114610
Region_state    115592
Latitude        113595
Longitude       113595
dtype: int64

Above shows the sum of null value for each column.

### 4. Drop columns “latitude” and “longitude” from the modified data set and save to a new data frame. Let’s call it Master_Clean.

In [18]:
Master_Clean = df.drop(['Latitude', 'Longitude'], axis=1)

In [19]:
Master_Clean.sample(2)

Job_id                Vertical    Company  \
189878  20285041307268219277898871714479  Information Technology  Bay State   
17224   22471426996576483087396855396900  Information Technology        NaN   

         Post_date Fill_date Time_to_fill  Salary      Location      City  \
189878  2017-02-22       NaN          NaN   87640  Beaufort, SC  Beaufort   
17224   2017-06-22       NaN          NaN   92764           NaN       NaN   

       State    Zip    County             Region_state  
189878    SC  29901  Beaufort  SC NONMETROPOLITAN AREA  
17224    NaN    NaN       NaN                      NaN

### 5. Using Master_Clean: Display jobs for the state “NY” that were posted in 2018.

In [20]:
Master_Clean.loc[(Master_Clean['Post_date'].str.contains('2018')) & (Master_Clean['State'] == 'NY')]

Job_id                Vertical  \
43      49166544275228979011485468005581  Information Technology   
56      32216824414038021036854127631301  Information Technology   
122     17686258734635581425901950904423  Information Technology   
284     68180000463505362721268597011820  Information Technology   
324     75506920553773445036596485697789  Information Technology   
363       467895701373045028283543226368  Information Technology   
401     50485279672009898454911121142589  Information Technology   
419     51333496372464413223171949419278  Information Technology   
430     18396944660472507504542277085363  Information Technology   
600     98550477161084188489066376128064  Information Technology   
622      5300000483730778286537489207643  Information Technology   
629     94516996256928574491330047487931  Information Technology   
643     48476158435811127247708931642112  Information Technology   
764     87504728153850971441130425901231  Information Technology   
1104     2696105939387916160564326211846  Information Technology   
1375    58723501807015432733721594581451  Information Technology   
1440      236521942553036418086001383896  Information Technology   
1449      415349121354810943135797503470  Information Technology   
1455    89393683620524664208128524249028  Information Technology   
1497      542917792728373007665722564775  Information Technology   
1561    48020264441568927672366851147667  Information Technology   
1572    55157641510439738974694495602806  Information Technology   
1593     9230481372292022845456755852312  Information Technology   
1725     2430770416515896965637198734168  Information Technology   
1814    17800043801809774750475896980863  Information Technology   
1929    43618288290019071415467283767645  Information Technology   
2064    20058100941873781076259371250836  Information Technology   
2122    67672646289841954190304222562626  Information Technology   
2149    71766127817365679338445621712580  Information Technology   
2162    79469812433179217818017160947095  Information Technology   
...                                  ...                     ...   
197255  18969907972565992937327855096596  Information Technology   
197361  36046790707551154563521215910714  Information Technology   
197368   8524828046409216745976823376079  Information Technology   
197384   1311719669786119554146956771515  Information Technology   
197564  50856440832349436360539868322162  Information Technology   
197880   8847539301471242419300630523660  Information Technology   
197910  94815303305634576837137950166525  Information Technology   
197992  50321414282363859021768866379287  Information Technology   
198024  39041548461717541373412730155378  Information Technology   
198189    569146260930962875804765510947  Information Technology   
198561   7471868774061666461522698576353  Information Technology   
198603   9818366192349086671944775892514  Information Technology   
198670   5382799510897294396466316554130  Information Technology   
198801  91732396453308685145715421293170  Information Technology   
198988   9359392156982846413247249713824  Information Technology   
199133  25898906526102308535443468941484  Information Technology   
199166  16305241635443566880606153397239  Information Technology   
199222  14064851454388698499979288682358  Information Technology   
199224   8620746492604713440117444191734  Information Technology   
199261  22951526576313531930218823616884  Information Technology   
199429  23195715127174399400005331847949  Information Technology   
199457  27263510140295428835718499697299  Information Technology   
199541  51615168366031172817008531041075  Information Technology   
199592    388342878600022764972914668644  Information Technology   
199618  85409671971447082949452856038518  Information Technology   
199631    146402745847645853174089044109  Information Technology   
199673  47938939537148512475744262642660  Information Technology   
199704   5741777954729

### 6. Using Master_Clean: Display jobs from all states other than the states “NY” and “CA” that offer a salary higher than 80,000. Do NOT include jobs that have NaN values for state column.

In [21]:
Master_Clean[(Master_Clean['State'] != 'NY') & (Master_Clean['State'] != 'CA') & (~Master_Clean['State'].isna())
            & (Master_Clean['Salary'] > 80000)]

Job_id                Vertical  \
4         490138684948094100363899533762  Information Technology   
5       74657902746882254946742123272434  Information Technology   
14      84400929153273312343387471187380  Information Technology   
23      73008818513451263769918234460534  Information Technology   
27      18130503948134891967593913939232  Information Technology   
31      92708943921819158278275137368013  Information Technology   
40       9661280239453334742826718781338  Information Technology   
42        406721259826597271166136103276  Information Technology   
53       7955450235772249401977199367187  Information Technology   
61      45230337933216733091735774621770  Information Technology   
70      32300234787209199670389747356731  Information Technology   
79        681377002829632837068544453736  Information Technology   
91       2294022611392523687558580960170  Information Technology   
98        479846345677436844332805134097  Information Technology   
106       585290221529020586396558449210  Information Technology   
111      7322556087579982411331394292574  Information Technology   
113     67896938593777969319816987857458  Information Technology   
117     97985120333738407291384064447686  Information Technology   
127      9262822776645422534243596775910  Information Technology   
132     76815795931597383535130997277332  Information Technology   
134     32050403263333033551530798576632  Information Technology   
140       841180897139267646091158867550  Information Technology   
152      4955411090799012533273354266631  Information Technology   
155       590453087687138472210015490856  Information Technology   
170      2232585180554128357930863730305  Information Technology   
172     13528279308638411373437076080331  Information Technology   
176     98892870724669616565619510880097  Information Technology   
178     88821272302728088490022733682714  Information Technology   
196     97867142837488402514277774467367  Information Technology   
205     93810100216751162694528269701067  Information Technology   
...                                  ...                     ...   
199811  66644845555063977759745221468072  Information Technology   
199817  28189986413015382193773485784060  Information Technology   
199823  67144701268990983306139084609661  Information Technology   
199824  23864138654917115586614382855586  Information Technology   
199832  48668818823964996597604582690728  Information Technology   
199841  20422894363981492072207094415903  Information Technology   
199842    991330966318838185426710314825  Information Technology   
199849   2931557538817070613863014241943  Information Technology   
199850  52793827212433104783507075445426  Information Technology   
199851  33721924422733421020094324184691  Information Technology   
199858  10743012780908565822524456853657  Information Technology   
199873  92803753748390589578403354402993  Information Technology   
199881  56998704510015908401183967054429  Information Technology   
199883   3326995720686802362414056060296  Information Technology   
199900  75611024286625471647966796410699  Information Technology   
199906  48567356716758431267715048358223  Information Technology   
199908  55611745814807494056851258400317  Information Technology   
199911   5240714260723471463959794980949  Information Technology   
199927  36014695188139188602550873685790  Information Technology   
199928  10352428307047742246659020218630  Information Technology   
199931  42645467574831485947934124947265  Information Technology   
199932  46298306566497354435376351819611  Information Technology   
199937  71150182893657383337814698793519  Information Technology   
199939  21611034267202896876834663680868  Information Technology   
199953  16097640182645324302974660355134  Information Technology   
199967  59727115888834976523614459530230  Information Technology   
199971  68059010730050456084819985225862  Information Technology   
199976  23453717852758

### 7. Using Master_Clean: How many jobs have NaN in the Time_to_fill column? Replace all these nulls (or NaN) with the median of this column. Display first 25 and last 25 entries.

In [22]:
# How many jobs have NaN in the Time_to_fill column?
Master_Clean['Time_to_fill'].isna().value_counts()

True     151118
False     48882
Name: Time_to_fill, dtype: int64

There are 151,118 NaN values in the Time_to_fill column

In [23]:
# Replace nulls with the median value
median = Master_Clean['Time_to_fill'].median()
Master_Clean['Time_to_fill'].fillna(median, inplace=True)

In [24]:
Master_Clean.sample(2)

Job_id                Vertical Company  \
120370  9721124821006560133990613975629  Information Technology     NaN   
79361    501523901514034925244875808034  Information Technology     NaN   

         Post_date Fill_date Time_to_fill  Salary Location City State  Zip  \
120370  2017-05-17       NaN           12   62840      NaN  NaN   NaN  NaN   
79361   2017-09-27       NaN           12   45837      NaN  NaN   NaN  NaN   

       County Region_state  
120370    NaN          NaN  
79361     NaN          NaN

### 8. Using Master_Clean: Display only company, posting date and salary of jobs that are listed for the state “NJ”. After filtering, exclude NaN from all columns

In [25]:
only_nj = Master_Clean.loc[Master_Clean['State'] == 'NJ']
only_nj[['Company', 'Post_date', 'Salary']].dropna()

Company   Post_date  Salary
248                                           Siemens AG  2018-03-21  117154
347                                        Stefanini Inc  2017-01-25  104023
425              On The Border Mexican Grill and Cantina  2017-08-02   17115
474                                                Ricoh  2018-03-15   39479
545                                           BNY Mellon  2016-10-28  104219
634                                       NFI Industries  2018-04-14   81348
724                                    Quest Diagnostics  2017-08-29   36549
833                                    Bed Bath & Beyond  2018-01-18  123018
916                                       APN Consulting  2018-07-17  109966
965                      Stewart's All American | Walker  2017-08-18   35169
974                           Partner's Consulting, Inc.  2017-01-19   86999
1035                              Mondelez International  2018-02-18   97201
1125                          Youth Consultation Service  2018-01-29   21086
1195                                  Bartlett Dairy Inc  2017-08-02   26575
1315             General Dynamics Information Technology  2018-07-11   73125
1334                                            Best Buy  2017-08-03   23932
1347                                           iconectiv  2016-08-20  116220
1522                          Cooper University Hospital  2018-07-04  133500
1541                                                KPMG  2018-02-17   82905
1568                               Hired by Matrix, Inc.  2017-08-17  102786
1688                                            Goodyear  2018-07-11   43320
1760                                              Leidos  2017-03-01   70362
1943                                       JCC Metrowest  2017-04-11   32516
2202                               Otsuka Pharmaceutical  2016-11-09   98599
2305                                      TSR Consulting  2017-08-29  101261
2428                     Open Systems Technologies, Inc.  2017-07-18   93779
2637                       Platinum Acorn solutions, LLC  2017-10-13  128767
2669                             Veterans Sourcing Group  2017-08-24   69115
2695                                          BNY Mellon  2018-03-09  107369
2718                                       Deutsche Bank  2016-09-02  116480
...                                                  ...         ...     ...
198182                     SHI International Corporation  2016-09-07  113373
198298                                      Ralph Lauren  2018-03-16  140397
198334                                         WALGREENS  2017-12-19   56739
198391                               Scarinci Hollenbeck  2018-02-06   48119
198562                                        ELLKAY LLC  2018-02-07  129393
198581                    St. Joseph's Healthcare System  2016-08-24   68675
198601                                Aftermath Services  2017-08-05   31845
198618                               Answer Human Assets  2018-01-29   69081
198624                               Smartlink, LLC (HQ)  2017-08-11  118550
198731                                   The Forum Group  2016-12-03   93696
198843                                Plastic Executives  2017-08-18  129917
198888                    Pinnacle Treatment Center KY-I  2018-03-23   43091
199019                New Jersey Institute of Technology  2018-05-03   68467
199023                                    ACT-Consulting  2017-05-31   88835
199060                             Smith Arnold Partners  2018-04-17   86318
199151                                         PDDN INC.  2017-08-15  164738
199184                                   JP Morgan Chase  2017-09-18  124990
199202                                              Citi  2018-02-22   95004
199226                                   Lockheed Martin  2018-05-01  110064
199271                              Integra LifeSciences  2018-06-04  127740
199319  Veterans Affairs, Veterans Health Adm

### 9. Using Master_Clean: Use the .isin() method to display company, posting date, time to fill and salaries of jobs in the states “NC”, “PA” or “TX”. After filtering, exclude NaN from all columns.

In [26]:
states_9 = ['NC', 'PA', 'TX']
Master_Clean.loc[Master_Clean['State'].isin(states_9)][['Company', 'Post_date', 'Time_to_fill', 'Salary']].dropna()

Company   Post_date  \
4                                             Mindteck  2018-02-20   
19                                             WakeMed  2018-07-20   
23                                Weidenhammer Systems  2017-12-19   
25                                   The Plastek Group  2018-05-05   
31                                              zodiac  2016-11-23   
72                     Securitas USA - Shared Services  2018-01-19   
74                          Edwards Aquifier Authority  2018-01-20   
90                              Mondelez International  2018-03-31   
91                 Texas A&M University Corpus Christi  2016-10-14   
99                                    BioBridge Global  2018-07-25   
113                        E.ON Climate and Renewables  2018-07-07   
141                                            Emerson  2018-03-03   
152                                    SMART IT People  2017-08-14   
159                                             Safran  2018-06-19   
172                                 Camber Corporation  2016-10-07   
176                                          Medtronic  2018-03-24   
188                            Adventist Health System  2018-07-12   
202                        Baylor Scott & White Health  2018-07-19   
208                           WellSpan Health Services  2016-12-04   
212                                     Common Threads  2017-08-04   
215                  Hargrove Engineers + Constructors  2018-07-14   
219                                   General Electric  2018-05-25   
229                 Ness Software Engineering Services  2018-03-04   
273                                        Airgas Inc.  2018-07-11   
275                                  Advocate Staffing  2018-02-06   
337                                           Mckesson  2016-12-20   
362                                             PACCAR  2017-01-19   
375                         Neighbors Emergency Center  2018-01-11   
404                                        DataMasters  2018-04-26   
407                          Global Process Automation  2016-10-21   
...                                                ...         ...   
199472                   Westinghouse Electric Company  2017-06-01   
199476                         Methodist Health System  2018-05-22   
199477                           CHI St. Luke’s Health  2018-05-12   
199486                                      Intermedix  2018-05-26   
199505                       MD Anderson Cancer Center  2018-06-28   
199508                                      AUTONATION  2018-07-24   
199510                                          Penske  2018-04-11   
199530                                        Sunpower  2016-10-04   
199531                             Access Staffing LLC  2017-08-05   
199543                                        SPX FLOW  2018-06-04   
199550                     Cape Fear Community College  2018-06-28   
199634                                     Wells Fargo  2018-01-10   
199752                    Advanced Micro Devices, Inc.  2016-10-18   
199777                                 A.C.Coy Company  2018-04-18   
199802                                        Hill-Rom  2017-01-06   
199818                     Climate Controls & Security  2016-10-01   
199850                               APAR Technologies  2017-08-04   
199872                                    Hanger, Inc.  2018-04-17   
199883                                 GlaxoSmithKline  2018-04-27   
199887              Cook Children's Health Care System  2017-03-04   
199892              University of North Texas: Faculty  2018-02-20   
199898                                 JP Morgan Chase  2018-04-07   
199903                      Hewlett Packard Enterprise  2018-01-28   
199911                      TriQuest Business Services  2016-10-26   
199922  North Carolina Office of State Human Resources  2017-08-07   
199928                               Compass Group USA  2

### 10. Using Master_Clean: Use the contains method to display city, state, county and regions of jobs that have “New York” or “Los Angeles-Long” in their region state value.

In [27]:
Master_Clean.loc[(Master_Clean['Region_state'].str.contains('New York', na=False)) | 
                 (Master_Clean['Region_state'].str.contains('Los Angeles-Long', na=False))][
    ['City', 'State', 'County', 'Region_state']]

City State       County  \
1                  Northridge    CA  Los Angeles   
47              New York City    NY     New York   
56              New York City    NY     New York   
71                   Melville    NY      Suffolk   
122             New York City    NY     New York   
137                    Irvine    CA       Orange   
222                  Pasadena    CA  Los Angeles   
268               Studio City    CA  Los Angeles   
284                   Bohemia    NY      Suffolk   
315             New York City    NY     New York   
324             New York City    NY     New York   
334                  Valencia    CA  Los Angeles   
347               Jersey City    NJ       Hudson   
349                Sun Valley    CA  Los Angeles   
350             New York City    NY     New York   
363             New York City    NY     New York   
394             New York City    NY     New York   
401                  Flushing    NY       Queens   
419             New York City    NY     New York   
425                   Paramus    NJ       Bergen   
430                  Brooklyn    NY        Kings   
462               Los Angeles    CA  Los Angeles   
474                    Newark    NJ        Essex   
517                 San Dimas    CA  Los Angeles   
535     Rolling Hills Estates    CA  Los Angeles   
545              Florham Park    NJ       Morris   
563             New York City    NY     New York   
600             New York City    NY     New York   
605               Los Angeles    CA  Los Angeles   
622             New York City    NY     New York   
...                       ...   ...          ...   
199284                  Azusa    CA  Los Angeles   
199314          New York City    NY     New York   
199319            East Orange    NJ        Essex   
199342                    NaN    CA       Orange   
199352             Bellflower    CA  Los Angeles   
199377                 Totowa    NJ      Passaic   
199457          New York City    NY     New York   
199460          New York City    NY     New York   
199461               Pasadena    CA  Los Angeles   
199470            Jersey City    NJ       Hudson   
199493                Burbank    CA  Los Angeles   
199507          New York City    NY     New York   
199541          New York City    NY     New York   
199563                 Newark    NJ        Essex   
199592          New York City    NY     New York   
199595               Harrison    NY  Westchester   
199618           White Plains    NY  Westchester   
199673          New York City    NY     New York   
199684            Los Angeles    CA  Los Angeles   
199704               Brooklyn    NY        Kings   
199715            Jersey City    NJ       Hudson   
199720            Jersey City    NJ       Hudson   
199730              La Mirada    CA  Los Angeles   
199743              Hollywood    CA  Los Angeles   
199811            Jersey City    NJ       Hudson   
199827              Santa Ana    CA       Orange   
199846          New York City    NY     New York   
199890            Stony Brook    NY      Suffolk   
199956           White Plains    NY  Westchester   
199981          New York City    NY     New York   

                                             Region_state  
1                Los Angeles-Long Beach-Santa Ana, CA MSA  
47      New York-Northern New Jersey-Long Island, NY-N...  
56      New York-Northern New Jersey-Long Island, NY-N...  
71      New York-Northern New Jersey-Long Island, NY-N...  
122     New York-Northern New Jersey-Long Island, NY-N...  
137              Los Angeles-Long Beach-Santa Ana, CA MSA  
222              Los Angeles-Long Beach-Santa Ana, CA MSA  
268              Los Angeles-Long Beach-Santa Ana, CA MSA  
284     New York-Northern New Jersey-Long Island, NY-N...  
315     New York-Northern New Jersey-Long Island, NY-N...  
324     New York-Northern New Jersey-Long Island, NY-N...  
334              Los Angeles-Long Beach-Santa Ana, CA MSA  
347     New York-Nor

### 11. Using Master_Clean: Display company, vertical, posting date and salaries for all jobs posted during the years 2017 and 2018 in “New York” or “Seattle” region states that had salaries higher than 80,000. Exclude NaN from company column.

In [28]:
Master_Clean.loc[((Master_Clean['Post_date'].str.contains('2017')) 
                 | (Master_Clean['Post_date'].str.contains('2018'))) 
                 & ((Master_Clean['Region_state'].str.contains('New York', na=False)) 
                 | (Master_Clean['Region_state'].str.contains('Seattle', na=False))) 
                 & (Master_Clean['Salary'] > 80000)][['Company', 'Vertical', 'Post_date', 'Salary']].dropna(subset=['Company'])

Company                Vertical  \
47                                          AIG  Information Technology   
56                                         MUFG  Information Technology   
61                                    Microsoft  Information Technology   
122                                         RBC  Information Technology   
280     General Dynamics Information Technology  Information Technology   
324            Federal Reserve Bank of New York  Information Technology   
347                               Stefanini Inc  Information Technology   
363                                 Wells Fargo  Information Technology   
374                                    T-Mobile  Information Technology   
394                                         RBC  Information Technology   
430                             JP Morgan Chase  Information Technology   
456                                    T-Mobile  Information Technology   
622                                Ralph Lauren  Information Technology   
629                         Cameron Craig Group  Information Technology   
649                                BCT Partners  Information Technology   
791                              Techlink, Inc.  Information Technology   
833                           Bed Bath & Beyond  Information Technology   
842                     U.S. Federal Government  Information Technology   
916                              APN Consulting  Information Technology   
1035                     Mondelez International  Information Technology   
1055                                 RSM US LLP  Information Technology   
1312                         Wildlife Computers  Information Technology   
1539                                  Microsoft  Information Technology   
1541                                       KPMG  Information Technology   
1561                            JP Morgan Chase  Information Technology   
1568                      Hired by Matrix, Inc.  Information Technology   
1652                               Point Inside  Information Technology   
1736                                  Microsoft  Information Technology   
1910                                  Microsoft  Information Technology   
1977                                      Fluke  Information Technology   
...                                         ...                     ...   
198624                      Smartlink, LLC (HQ)  Information Technology   
198670                          Custom Staffing  Information Technology   
198678                                Bloomberg  Information Technology   
198693                                Microsoft  Information Technology   
198750             Engineering Consultancy Firm  Information Technology   
198801                          JP Morgan Chase  Information Technology   
198843                       Plastic Executives  Information Technology   
198988                                     KPMG  Information Technology   
199023                           ACT-Consulting  Information Technology   
199060                    Smith Arnold Partners  Information Technology   
199101                                      REI  Information Technology   
199133                           Wolters Kluwer  Information Technology   
199151                                PDDN INC.  Information Technology   
199184                          JP Morgan Chase  Information Technology   
199202                                     Citi  Information Technology   
199261                                      GTT  Information Technology   
199271                     Integra LifeSciences  Information Technology   
199314                    School of Visual Arts  Information Technology   
199457                              e-Frontiers  Information Technology   
199470                              BNP Paribas  Information Technology   
199478                                Microsoft  Information Technology   
199483                  Greenpoint Technologies  Information Tech

### 12. Using Master_Clean: Calculate the percentage of jobs offering a six-figure salary or higher in the state “NY” in 2018.

In [29]:
six_fig_NY_2018 = Master_Clean.loc[(Master_Clean['Salary'] >= 100000) & (Master_Clean['State'] == 'NY') 
                 & (Master_Clean['Post_date'].str.contains('2018'))]

all_NY_2018 = Master_Clean.loc[(Master_Clean['State'] == 'NY') & (Master_Clean['Post_date'].str.contains('2018'))]

In [30]:
(len(six_fig_NY_2018) / len(all_NY_2018)) * 100

37.27637824023366

Jobs offering a sex-figures salary or higher in NY in 2018 was 37% of all jobs in NY in 2018.

### 13. Using Master_Clean: Calculate the percentage of job postings for the company “Wells Fargo” where the salary offered is above the mean of all salaries in the states of “NY” or “CA”.

In [31]:
# The mean of all salaries in the states of 'NY' or 'CA'
mean_sal_nyca = Master_Clean.loc[(Master_Clean['State'] == 'NY') 
                                 | (Master_Clean['State'] == 'CA')].agg({'Salary':'mean'}).item()
mean_sal_nyca

83551.04565557597

In [33]:
# The percentage of job postings for the company "Wells Fargo", of all jobs
jobs_wf_nyca = Master_Clean.loc[(Master_Clean['Company'] == 'Wells Fargo') & (Master_Clean['Salary'] > mean_sal_nyca)]
jobs_wf = Master_Clean.loc[Master_Clean['Company'] == 'Wells Fargo']

In [34]:
(len(jobs_wf_nyca) / len(jobs_wf)) * 100

73.78516624040921

The percentage of Wells Fargo job postings that are above the mean of all salaries in NY or CA, is about 74% of all Wells Fargo job postings.